### ISSUE TITLE
simDetectorIOC crashes at startup on ubuntu 24.04.1 LTS

### ISSUE THREAD

**Issue by LeeYangLBLBCS** [areaDetector/ADSimDetector#26]
````
IOC crashes when running in non-debug mode.
Though it does NOT crash in debug mode.

lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04.1 LTS
Release:	24.04
Codename:	noble

/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/iocBoot/iocSimDetector$ ../../bin/linux-x86_64/simDetectorApp  st.cmd.linux 
< envPaths.linux
epicsEnvSet("IOC","iocSimDetector")
epicsEnvSet("TOP","/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC")
epicsEnvSet("ADSIMDETECTOR","/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/../..")
epicsEnvSet("SUPPORT","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support")
epicsEnvSet("ASYN","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/asyn-R4-44-2")
epicsEnvSet("AREA_DETECTOR","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7")
epicsEnvSet("ADSUPPORT","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport")
epicsEnvSet("ADCORE","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore")
epicsEnvSet("AUTOSAVE","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/autosave-R5-10")
epicsEnvSet("BUSY","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/busy-R1-7-2")
epicsEnvSet("CALC","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/calc-R3-7-4")
epicsEnvSet("SNCSEQ","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/seq-2-2-9")
epicsEnvSet("SSCAN","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/sscan-R2-11-3")
epicsEnvSet("DEVIOCSTATS","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/iocStats-3-1-16")
epicsEnvSet("STD","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/std-R3-6-1")
epicsEnvSet("EPICS_BASE","/opt/epics/base-7.0.4")
< st_base.cmd
# Must have loaded envPaths via st.cmd*
errlogInit(20000)
dbLoadDatabase("/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/dbd/simDetectorApp.dbd")
simDetectorApp_registerRecordDeviceDriver(pdbbase) 
# Prefix for all records
epicsEnvSet("PREFIX", "13SIM1:")
# The port name for the detector
epicsEnvSet("PORT",   "SIM1")
# The queue size for all plugins
epicsEnvSet("QSIZE",  "20")
# The maximum image width; used to set the maximum size for this driver and for row profiles in the NDPluginStats plugin
epicsEnvSet("XSIZE",  "1024")
# The maximum image height; used to set the maximum size for this driver and for column profiles in the NDPluginStats plugin
epicsEnvSet("YSIZE",  "1024")
# The maximum number of time series points in the NDPluginStats plugin
epicsEnvSet("NCHANS", "2048")
# The maximum number of frames buffered in the NDPluginCircularBuff plugin
epicsEnvSet("CBUFFS", "500")
# The maximum number of threads for plugins which can run in multiple threads
epicsEnvSet("MAX_THREADS", "8")
# The search path for database files
epicsEnvSet("EPICS_DB_INCLUDE_PATH", "/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore/db")
asynSetMinTimerPeriod(0.001)
asynSetMinTimerPeriod is not currently supported on this OS
# The EPICS environment variable EPICS_CA_MAX_ARRAY_BYTES needs to be set to a value at least as large
# as the largest image that the standard arrays plugin will send.
# That value is $(XSIZE) * $(YSIZE) * sizeof(FTVL data type) for the FTVL used when loading the NDStdArrays.template file.
# The variable can be set in the environment before running the IOC or it can be set here.
# It is often convenient to set it in the environment outside the IOC to the largest array any client 
# or server will need.  For example 10000000 (ten million) bytes may be enough.
# If it is set here then remember to also set it outside the IOC for any CA clients that need to access the waveform record.  
# Do not set EPICS_CA_MAX_ARRAY_BYTES to a value much larger than that required, because EPICS Channel Access actually
# allocates arrays of this size every time it needs a buffer larger than 16K.
# Uncomment the following line to set it in the IOC.
#epicsEnvSet("EPICS_CA_MAX_ARRAY_BYTES", "10000000")
# Create a simDetector driver
# simDetectorConfig(const char *portName, int maxSizeX, int maxSizeY, int dataType,
#                   int maxBuffers, int maxMemory, int priority, int stackSize)
simDetectorConfig("SIM1", 1024, 1024, 1, 0, 0)
# To have the rate calculation use a non-zero smoothing factor use the following line
#dbLoadRecords("simDetector.template",     "P=$(PREFIX),R=cam1:,PORT=$(PORT),ADDR=0,TIMEOUT=1,RATE_SMOOTH=0.2")
dbLoadRecords("/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/../../db/simDetector.template","P=13SIM1:,R=cam1:,PORT=SIM1,ADDR=0,TIMEOUT=1")
*** buffer overflow detected ***: terminated
Aborted (core dumped)

**Comment by MarkRivers** [comment]
You got a core dump, so you should run gdb on that, and give the “backtrace” command to see the stack trace. 

It appears that you are using areaDetector R3-7, which is 4.5 years old. Is there a reason for running that old version?

**Comment by LeeYangLBLBCS** [comment]
````
Can I get anything useful with gdb on a "non debug" executable?

I've been using areaDetector R3-7 since a few years ago, but some components are fairly new.  And everything still works fine with Ubuntu 22.0.4 LTS and recent versions of Rocky Linux, so I havn't had the need to go to the latest version.  But may be that's what I should do now.

e.g. ADSimDetector is R2-10, from the RELEASE.md file:
R2-10 (October 22, 2019)
=========================
* Added support for NDArray datatypes NDInt64 and NDUInt64
  * This requires asyn R4-37 and ADCore R3-8
* Added iocSimDetector/testHDF5Compression.py to test HDF5 compression.
* Added Time64 attribute to iocSimDetector/simDetectorAttributes.xml.

Also, fairly new Asyn: RELEASE.md:
## Release 4-45 (May XXX, 2023)
- devEpics
  - Changed devAsynXXXArray::interruptCallback so it returns immediately if interruptAccept is still 0, i.e. before iocInit
    is mostly complete.  This was causing warnings if callbacks occured before iocInit.

**Comment by MarkRivers** [comment]
Yes, non-debug builds on Linux still have lots of useful information for gdb.

I am surprised that an OS change would cause this problem. Did you rebuild everything for this OS?

**Comment by LeeYangLBLBCS** [comment]
Yes, I rebuilt everything, starting from epics base 7.0.4.
The motorMotorSim IOC came up without any problems, though I havn't tested anything else beside the simulated IOCs so far.

**Comment by LeeYangLBLBCS** [comment]
````
by the way, I just got the latest ADSimDetector, and had the same core dump. Here is the git repo info of the latest version I got:
git log
commit 384f9ea9e5f574ac5ccdef6b0a898dad21009601 (HEAD -> master, origin/master, origin/HEAD)
Author: Mark Rivers <rivers@cars.uchicago.edu>
Date:   Thu Nov 14 17:15:08 2024 -0600

    New file to generate attributes of all numeric datatypes

commit ae54f7ee9d5586d76a1baf018b2ba32a0ca055f4 (tag: R2-11)
Author: Mark Rivers <rivers@cars.uchicago.edu>
Date:   Sun Apr 7 17:36:00 2024 -0500

    Notes for R2-11

commit 2eea42549864df8c6068fc5dc18d93ba37493eb1
Author: Mark Rivers <rivers@cars.uchicago.edu>
Date:   Sun Apr 7 17:33:48 2024 -0500

    Fix typo

commit ef63722aa6f93243392268257a7168f0e9742604
Author: Mark Rivers <rivers@cars.uchicago.edu>
Date:   Sun Apr 7 17:32:26 2024 -0500

**Comment by MarkRivers** [comment]
You can also just run ADSimDetector under gdb, and the type “run st.cmd”. 

Perhaps some a newer compiler is exposing a problem we have not seen before.  What version of gcc is this?

**Comment by LeeYangLBLBCS** [comment]
````
GCC version is 13.3.0, here is the output for gcc -v:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-linux-gnu/13/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 13.3.0-6ubuntu2~24.04' --with-bugurl=file:///usr/share/doc/gcc-13/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-13 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/libexec --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-libstdcxx-backtrace --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-13-fG75Ri/gcc-13-13.3.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-13-fG75Ri/gcc-13-13.3.0/debian/tmp-gcn/usr --enable-offload-defaulted --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 13.3.0 (Ubuntu 13.3.0-6ubuntu2~24.04)

**Comment by LeeYangLBLBCS** [comment]
````
here is the gdb start up log, up to the core dump:
lyang@lyHomeLinux:/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/iocBoot/iocSimDetector$ gdb ../../bin/linux-x86_64/simDetectorApp 
GNU gdb (Ubuntu 15.0.50.20240403-0ubuntu1) 15.0.50.20240403-git
Copyright (C) 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ../../bin/linux-x86_64/simDetectorApp...

This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.ubuntu.com>
Enable debuginfod for this session? (y or [n]) y
Debuginfod has been enabled.
To make this setting permanent, add 'set debuginfod enabled on' to .gdbinit.
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/bin/linux-x86_64/simDetectorApp
(No debugging symbols found in ../../bin/linux-x86_64/simDetectorApp)                                                                             
(gdb) run st.cmd.linux
Starting program: /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/bin/linux-x86_64/simDetectorApp st.cmd.linux
Downloading separate debug info for system-supplied DSO at 0x7ffff7fc3000
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/lib/linux-x86_64/libsimDetector.so
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore/lib/linux-x86_64/libNDPlugin.so        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore/lib/linux-x86_64/libADBase.so          
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libpvAccessIOC.so.7.1.1                                                
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/asyn-R4-44-2/lib/linux-x86_64/libasyn.so                        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/autosave-R5-10/lib/linux-x86_64/libautosave.so                  
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/busy-R1-7-2/lib/linux-x86_64/libbusy.so                         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/calc-R3-7-4/lib/linux-x86_64/libcalc.so                         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/iocStats-3-1-16/lib/linux-x86_64/libdevIocStats.so              
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/sscan-R2-11-3/lib/linux-x86_64/libsscan.so                      
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libdbRecStd.so.3.18.0                                                  
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0                                                    
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libCom.so.3.18.0                                                       
[Thread debugging using libthread_db enabled]                                                                                                     
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libnanohttp_stream.so
Downloading separate debug info for /lib/x86_64-linux-gnu/libstdc++.so.6                                                                          
Downloading separate debug info for /lib/x86_64-linux-gnu/libgcc_s.so.1                                                                           
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore/lib/linux-x86_64/libntndArrayConverter.so
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libnt.so.6.0.1                                                         
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libpvDatabase.so.4.5.2                                                 
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libnetCDF.so       
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libMagick++.so     
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libtiff.so         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libNeXus.so        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libhdf5.so         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libhdf5_hl.so      
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libbitshuffle.so   
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libblosc.so        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libjpeg.so         
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libca.so.4.13.6                                                        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libxml2.so         
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libpvAccess.so.7.1.1                                                   
Downloading separate debug info for /opt/epics/base-7.0.4/lib/linux-x86_64/libpvData.so.8.0.2                                                     
Downloading separate debug info for /lib/x86_64-linux-gnu/libtirpc.so.3                                                                           
Downloading separate debug info for /lib/x86_64-linux-gnu/libreadline.so.8                                                                        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libzlib.so         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libMagick.so       
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libcoders.so       
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libszip.so         
Downloading separate debug info for /lib/x86_64-linux-gnu/libgssapi_krb5.so.2                                                                     
Downloading separate debug info for /lib/x86_64-linux-gnu/libtinfo.so.6                                                                           
Downloading separate debug info for /lib/x86_64-linux-gnu/libtinfo.so.6                                                                           
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libbzlib.so        
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/liblcms.so         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libttf.so          
Downloading separate debug info for /lib/x86_64-linux-gnu/libXext.so.6                                                                            
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libjbig.so         
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libjp2.so          
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libpng.so          
Downloading separate debug info for /opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport/lib/linux-x86_64/libwebp.so         
Downloading separate debug info for /lib/x86_64-linux-gnu/libkrb5.so.3                                                                            
Downloading separate debug info for /lib/x86_64-linux-gnu/libk5crypto.so.3                                                                        
Downloading separate debug info for /lib/x86_64-linux-gnu/libcom_err.so.2                                                                         
Downloading separate debug info for /lib/x86_64-linux-gnu/libkrb5support.so.0                                                                     
Downloading separate debug info for /lib/x86_64-linux-gnu/libX11.so.6                                                                             
Downloading separate debug info for /lib/x86_64-linux-gnu/libkeyutils.so.1                                                                        
Downloading separate debug info for /lib/x86_64-linux-gnu/libxcb.so.1                                                                             
Downloading separate debug info for /lib/x86_64-linux-gnu/libXau.so.6                                                                             
Downloading separate debug info for /lib/x86_64-linux-gnu/libXdmcp.so.6                                                                           
Downloading separate debug info for /lib/x86_64-linux-gnu/libbsd.so.0                                                                             
Downloading separate debug info for /lib/x86_64-linux-gnu/libmd.so.0                                                                              
[New Thread 0x7ffff5a006c0 (LWP 2305631)]                                                                                                         
[Thread 0x7ffff5a006c0 (LWP 2305631) exited]
< envPaths.linux
epicsEnvSet("IOC","iocSimDetector")
epicsEnvSet("TOP","/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC")
epicsEnvSet("ADSIMDETECTOR","/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/../..")
epicsEnvSet("SUPPORT","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support")
epicsEnvSet("ASYN","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/asyn-R4-44-2")
epicsEnvSet("AREA_DETECTOR","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7")
epicsEnvSet("ADSUPPORT","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSupport")
epicsEnvSet("ADCORE","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore")
epicsEnvSet("AUTOSAVE","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/autosave-R5-10")
epicsEnvSet("BUSY","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/busy-R1-7-2")
epicsEnvSet("CALC","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/calc-R3-7-4")
epicsEnvSet("SNCSEQ","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/seq-2-2-9")
epicsEnvSet("SSCAN","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/sscan-R2-11-3")
epicsEnvSet("DEVIOCSTATS","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/iocStats-3-1-16")
epicsEnvSet("STD","/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/std-R3-6-1")
epicsEnvSet("EPICS_BASE","/opt/epics/base-7.0.4")
[New Thread 0x7ffff5bab6c0 (LWP 2305632)]
< st_base.cmd
# Must have loaded envPaths via st.cmd*
errlogInit(20000)
dbLoadDatabase("/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/dbd/simDetectorApp.dbd")
simDetectorApp_registerRecordDeviceDriver(pdbbase) 
# Prefix for all records
epicsEnvSet("PREFIX", "13SIM1:")
# The port name for the detector
epicsEnvSet("PORT",   "SIM1")
# The queue size for all plugins
epicsEnvSet("QSIZE",  "20")
# The maximum image width; used to set the maximum size for this driver and for row profiles in the NDPluginStats plugin
epicsEnvSet("XSIZE",  "1024")
# The maximum image height; used to set the maximum size for this driver and for column profiles in the NDPluginStats plugin
epicsEnvSet("YSIZE",  "1024")
# The maximum number of time series points in the NDPluginStats plugin
epicsEnvSet("NCHANS", "2048")
# The maximum number of frames buffered in the NDPluginCircularBuff plugin
epicsEnvSet("CBUFFS", "500")
# The maximum number of threads for plugins which can run in multiple threads
epicsEnvSet("MAX_THREADS", "8")
# The search path for database files
epicsEnvSet("EPICS_DB_INCLUDE_PATH", "/opt/epics/base-7.0.4/../modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADCore/db")
asynSetMinTimerPeriod(0.001)
asynSetMinTimerPeriod is not currently supported on this OS
# The EPICS environment variable EPICS_CA_MAX_ARRAY_BYTES needs to be set to a value at least as large
# as the largest image that the standard arrays plugin will send.
# That value is $(XSIZE) * $(YSIZE) * sizeof(FTVL data type) for the FTVL used when loading the NDStdArrays.template file.
# The variable can be set in the environment before running the IOC or it can be set here.
# It is often convenient to set it in the environment outside the IOC to the largest array any client 
# or server will need.  For example 10000000 (ten million) bytes may be enough.
# If it is set here then remember to also set it outside the IOC for any CA clients that need to access the waveform record.  
# Do not set EPICS_CA_MAX_ARRAY_BYTES to a value much larger than that required, because EPICS Channel Access actually
# allocates arrays of this size every time it needs a buffer larger than 16K.
# Uncomment the following line to set it in the IOC.
#epicsEnvSet("EPICS_CA_MAX_ARRAY_BYTES", "10000000")
# Create a simDetector driver
# simDetectorConfig(const char *portName, int maxSizeX, int maxSizeY, int dataType,
#                   int maxBuffers, int maxMemory, int priority, int stackSize)
simDetectorConfig("SIM1", 1024, 1024, 1, 0, 0)
[New Thread 0x7ffff5b2a6c0 (LWP 2305637)]
[New Thread 0x7ffff51ff6c0 (LWP 2305638)]
[New Thread 0x7ffff50fe6c0 (LWP 2305640)]
[New Thread 0x7ffff4ffd6c0 (LWP 2305641)]
[New Thread 0x7ffff4efc6c0 (LWP 2305642)]
# To have the rate calculation use a non-zero smoothing factor use the following line
#dbLoadRecords("simDetector.template",     "P=$(PREFIX),R=cam1:,PORT=$(PORT),ADDR=0,TIMEOUT=1,RATE_SMOOTH=0.2")
dbLoadRecords("/opt/epics/modules/synApps_6_1_epics7/support/areaDetector-R3-7/ADSimDetector/iocs/simDetectorIOC/../../db/simDetector.template","P=13SIM1:,R=cam1:,PORT=SIM1,ADDR=0,TIMEOUT=1")
*** buffer overflow detected ***: terminated

Thread 1 "simDetectorApp" received signal SIGABRT, Aborted.
Download failed: Invalid argument.  Continuing without source file ./nptl/./nptl/pthread_kill.c.
__pthread_kill_implementation (no_tid=0, signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:44
warning: 44	./nptl/pthread_kill.c: No such file or directory
(gdb)

**Comment by MarkRivers** [comment]
Once you get the gdb prompt after the crash you need to type “backtrace” or “bt”. Then we can see the call stack.

**Comment by LeeYangLBLBCS** [comment]
````
continuing from the previous screen output, bt gives the following:
Thread 1 "simDetectorApp" received signal SIGABRT, Aborted.
Download failed: Invalid argument.  Continuing without source file ./nptl/./nptl/pthread_kill.c.
__pthread_kill_implementation (no_tid=0, signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:44
warning: 44	./nptl/pthread_kill.c: No such file or directory
(gdb) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=<optimized out>, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007ffff784526e in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007ffff78288ff in __GI_abort () at ./stdlib/abort.c:79
#5  0x00007ffff78297b6 in __libc_message_impl (fmt=fmt@entry=0x7ffff79ce765 "*** %s ***: terminated\n") at ../sysdeps/posix/libc_fatal.c:132
#6  0x00007ffff7936c19 in __GI___fortify_fail (msg=msg@entry=0x7ffff79ce74c "buffer overflow detected") at ./debug/fortify_fail.c:24
#7  0x00007ffff79365d4 in __GI___chk_fail () at ./debug/chk_fail.c:28
#8  0x00007ffff7c2dc29 in dbAllocRecord () from /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0
#9  0x00007ffff7c22573 in dbCreateRecord () from /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0
#10 0x00007ffff7c29413 in dbRecordHead () from /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0
#11 0x00007ffff7c2ab2d in dbReadCOM () from /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0
#12 0x00007ffff7bf57eb in dbLoadRecords () from /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0
#13 0x00007ffff7c1cdf4 in dbLoadRecordsCallFunc () from /opt/epics/base-7.0.4/lib/linux-x86_64/libdbCore.so.3.18.0
#14 0x00007ffff7b8665c in iocshBody () from /opt/epics/base-7.0.4/lib/linux-x86_64/libCom.so.3.18.0
#15 0x00007ffff7b85db3 in iocshBody () from /opt/epics/base-7.0.4/lib/linux-x86_64/libCom.so.3.18.0
#16 0x000055555555f526 in main ()
(gdb)

**Comment by MarkRivers** [comment]
This looks like a problem in EPICS base. I suggest checking for known issues is base 7.0.4 or just trying 7.0.8.

**Comment by LeeYangLBLBCS** [comment]
sounds good. I'll try the newer base.

**Comment by MarkRivers** [comment]
I don’t think we should close this until it is solved.
Reopening

**Comment by anjohnson** [comment]
I think you will need to use EPICS 7.0.8.1 as that includes a workaround needed for recent versions of gcc/glibc. The 7.0 branch of epics-base includes proper fixes, but haven't made it into a released version of Base yet so 7.0.8.1 is what I recommend.

**Comment by LeeYangLBLBCS** [comment]
thanks for the help.
Indeed, the latest base, 7.0.8.1, fixed the issue.
Though I have unrelated issues with a few other modules, which I'll address in other threads.
